### PR TITLE
[WIP] Documentation refactoring

### PR DESCRIPTION
**Pull request by m0mus**
Changed documentation structure
SE and MP docs separated

### CODE REVIEW

**Comment by m0mus** [comment]
Related to https://github.com/oracle/helidon/issues/1515

**Comment by romain-grecourt** [comment]
The docs have been flattened, there is no `src/main/docs/` directory anymore.

**Comment by m0mus** [comment]
@romain-grecourt What do you mean? It's here: https://github.com/m0mus/helidon/tree/docs_refactoring/docs/src/main/docs

**Comment by m0mus** [comment]
It's been rebased and should be mergeable now.

File: docs/mp/extensions/02_cdi_datasource-ucp.adoc [hunk 1]
```diff
@@ -88,4 +88,4 @@ are parsed, and the remaining portion of each name is treated as a
 https://docs.oracle.com/javase/tutorial/javabeans/writing/properties.html[Java
 Bean property] of the
 https://docs.oracle.com/en/database/oracle/oracle-database/19/jjuar/oracle/ucp/jdbc/PoolDataSource.html[`oracle.ucp.jdbc.PoolDataSource`]
-class.
+class.
```
Comment (romain-grecourt) [inline docs/mp/extensions/02_cdi_datasource-ucp.adoc]: This looks like a white space change. This should be displayed in the diff as "File renamed without changes"

File: docs/guides/30_dockerfile.adoc [hunk 2]
```diff
@@ -223,4 +223,4 @@ RUN ["jlink", "--compress=2", "--strip-debug", "--no-header-files", \
      "--add-modules", \
      "java.base,java.logging,java.sql,java.desktop,java.management,java.naming,jdk.unsupported", \
      "--output", "/var/tmp/myjre"]
-----
+----
```
Comment (romain-grecourt) [inline docs/guides/30_dockerfile.adoc]: This looks like a white space change. This should be displayed in the diff as "File renamed without changes"

File: docs/mp/restclient/09_rest-client.adoc [hunk 3]
```diff
@@ -113,4 +113,4 @@ Then create runnable method the same way as it is described in
 `Creating new client - Interface`,but with baseUri `http://localhost:8080/greet`
 and the above interface.
 
-By calling `GreetRestClient.getDefaultMessage()` you reach the endpoint of Helidon quickstart.
+By calling `GreetRestClient.getDefaultMessage()` you reach the endpoint of Helidon quickstart.
```
Comment (romain-grecourt) [inline docs/mp/restclient/09_rest-client.adoc]: This looks like a white space change. This should be displayed in the diff as "File renamed without changes"

File: docs/mp/jaxrs/10_reactive-routing.adoc [hunk 4]
```diff
@@ -121,4 +121,4 @@ io.helidon.examples.AdminService:
     required: true
   routing-path:
     path: "/management"
-----
+----
```
Comment (romain-grecourt) [inline docs/mp/jaxrs/10_reactive-routing.adoc]: This looks like a white space change. This should be displayed in the diff as "File renamed without changes"

File: docs/se/health/01_health.adoc [hunk 5]
```diff
@@ -251,4 +251,4 @@ values. Helidon reports both of these to maintain backward compatibility with ol
 versions of MicroProfile Health. This behavior can be disabled by setting
 the property `health.backward-compatible` to `false`, in which case only `"status"`
 is reported. Future versions of Helidon will drop support for older versions of Health,
-so it is recommended to rely on `"status"` instead of `"outcome"` in your applications.
+so it is recommended to rely on `"status"` instead of `"outcome"` in your applications.
```
Comment (romain-grecourt) [inline docs/se/health/01_health.adoc]: This looks like a white space change. This should be displayed in the diff as "File renamed without changes"

File: docs/se/security/05_extensibility.adoc [hunk 6]
```diff
@@ -129,4 +129,4 @@ the integration component attempts to authenticate the request.
 
 Because other components of Helidon Security (such as ABAC validators) query the request for annotations,
 the integration component should also collect all annotations from the resource and correctly configure
-them when creating the security request.
+them when creating the security request.
```
Comment (romain-grecourt) [inline docs/se/security/05_extensibility.adoc]: This looks like a white space change. This should be displayed in the diff as "File renamed without changes"

File: docs/se/webserver/11_access-log.adoc [hunk 7]
```diff
@@ -166,4 +166,4 @@ io.helidon.webserver.accesslog.AccessLogHandler.append=true
 io.helidon.webserver.AccessLog.level=INFO
 io.helidon.webserver.AccessLog.useParentHandlers=false
 io.helidon.webserver.AccessLog.handlers=io.helidon.webserver.accesslog.AccessLogHandler
-----
+----
```
Comment (romain-grecourt) [inline docs/se/webserver/11_access-log.adoc]: This looks like a white space change. This should be displayed in the diff as "File renamed without changes"

**Comment by m0mus** [comment]
@romain-grecourt You can hide white space changes in your diff settings. https://github.blog/2018-05-01-ignore-white-space-in-code-review/

**Pull request merged**